### PR TITLE
Refactor types commands to use API

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Refactor to use proper API instead of the `rc_core_cmd*()` calls in Visual mode and some of the Types commands.
Also:
- Fixed the bug when running `afck`
- Refactored to use `RzOutputMode` enum instead of the `int`/`char` as the output mode in some flag commands.

**Test plan**

CI is green

### CODE REVIEW

**Review by wargio — APPROVED** [review]

File: librz/core/visual.c [hunk 1]
```diff
@@ -3235,7 +3235,8 @@ RZ_API int rz_core_visual_cmd(RzCore *core, const char *arg) {
 				ut64 addr = rz_debug_reg_get(core->dbg, "PC");
 				if (addr && addr != UT64_MAX) {
 					rz_core_seek_and_save(core, addr, true);
-					rz_core_cmdf(core, "ar `arn PC`=0x%" PFMT64x, addr);
```
Comment (ret2libc) [inline librz/core/visual.c:3238]: I think this line was written some time ago... you can now do `ar PC=0xcafe` and it works.
I think you can directly do `rz_core_analysis_set_reg(core, "PC", addr)`.

File: test/db/cmd/cmd_tcc [hunk 2]
```diff
@@ -34,17 +34,17 @@ tccl
 tcc*
 EOF
 EXPECT=<<EOF
-["rax r13.swift (rdi, rsi, rdx, rcx, r8, r9, xmm0, xmm1, xmm2, xmm3, xmm4) r12;","rax amd64 (rdi, rsi, rdx, rcx, r8, r9, xmm0, xmm1, xmm2, xmm3, xmm4);","rax amd64syscall (rdi, rsi, rdx, r10, r8, r9);","rax ms (rcx, rdx, r8, r9, stack);","rdi reg (rdi, rsi, rdx, rcx);"]
-rax r13.swift (rdi, rsi, rdx, rcx, r8, r9, xmm0, xmm1, xmm2, xmm3, xmm4) r12;
+["rax amd64 (rdi, rsi, rdx, rcx, r8, r9, xmm0, xmm1, xmm2, xmm3, xmm4);","rax amd64syscall (rdi, rsi, rdx, r10, r8, r9);","rax ms (rcx, rdx, r8, r9, stack);","rdi reg (rdi, rsi, rdx, rcx);","rax r13.swift (rdi, rsi, rdx, rcx, r8, r9, xmm0, xmm1, xmm2, xmm3, xmm4) r12;"]
```
Comment (ret2libc) [inline test/db/cmd/cmd_tcc:37]: why this change? I wouldn't expect any difference

Comment (XVilka) [reply test/db/cmd/cmd_tcc:37]: It is now sorted alphabetically, before it wasn't sorted, it seems how `sdb_foreach_list()` works.

File: librz/core/visual.c (hunk 1 above)
Comment (XVilka) [reply librz/core/visual.c:3238]: Right, changed to the direct call.

**Review by ret2libc — APPROVED** [review]

File: librz/core/visual.c (hunk 1 above)
Comment (XVilka) [reply librz/core/visual.c:3238]: Right, changed to the direct call.
